File: jdroid-android-firebase-remoteconfig/src/debug/java/com/jdroid/android/firebase/remoteconfig/FirebaseRemoteConfigDebugAppLifecycleCallback.java
package com.jdroid.android.firebase.remoteconfig;

import android.content.Context;
import android.support.annotation.NonNull;

import com.jdroid.android.application.AbstractApplication;
import com.jdroid.android.debug.DebugSettingsHelper;
import com.jdroid.android.lifecycle.ApplicationLifecycleCallback;

public class FirebaseRemoteConfigDebugAppLifecycleCallback extends ApplicationLifecycleCallback {
	
	@Override
	public void onProviderInit(Context context) {
		
		if (MockRemoteConfigLoader.isMocksEnabled()) {
			AbstractApplication.get().setRemoteConfigLoader(new MockRemoteConfigLoader(true));
		}
		
		DebugSettingsHelper.addPreferencesAppender(new FirebaseRemoteConfigPrefsAppender());
	}
	
	@NonNull
	@Override
	public Integer getInitOrder() {
		return 1;
	}
	
}

File: jdroid-android-firebase-remoteconfig/src/debug/java/com/jdroid/android/firebase/remoteconfig/FirebaseRemoteConfigDebugAppLifecycleCallback.kt
package com.jdroid.android.firebase.remoteconfig

import android.content.Context

import com.jdroid.android.application.AbstractApplication
import com.jdroid.android.debug.DebugSettingsHelper
import com.jdroid.android.lifecycle.ApplicationLifecycleCallback

class FirebaseRemoteConfigDebugAppLifecycleCallback : ApplicationLifecycleCallback() {

    override fun onProviderInit(context: Context) {

        if (MockRemoteConfigLoader.isMocksEnabled()) {
            AbstractApplication.get().remoteConfigLoader = MockRemoteConfigLoader(true)
        }

        DebugSettingsHelper.addPreferencesAppender(FirebaseRemoteConfigPrefsAppender())
    }

    override fun getInitOrder(): Int {
        return 1
    }
}
